examples/plot_learn_from_imbalanced_dataset.py

```diff
@@ -0,0 +1,276 @@
+"""
+==============================
+Learn from imbalanced datasets
+==============================
+
+This example illustrates the problem induced by learning on datasets having
+imbalanced classes. Subsequently, we compare different approaches alleviating
+these negative effects using scikit-learn.
+
+"""
+
+# Authors: Guillaume Lemaitre <g.lemaitre58@gmail.com>
+# License: MIT
+
+print(__doc__)
+
+###############################################################################
+# Problem definition
+###############################################################################
+
+from sklearn.datasets import fetch_openml
+
+df, y = fetch_openml('adult', version=2, as_frame=True, return_X_y=True)
+# we are dropping the following features:
+# - "fnlwgt": this feature was created while studying the "adult" dataset.
+#   Thus, we will not use this feature which is not acquired during the survey.
+# - "education-num": it is encoding the same information than "education".
+#   Thus, we are removing one of these 2 features.
+df = df.drop(columns=['fnlwgt', 'education-num'])
+
+###############################################################################
+# The "adult" dataset as a class ratio of about 3:1
+
+classes_count = y.value_counts()
+classes_count
+
+###############################################################################
+# This dataset is only slightly imbalanced. To better highlight the effect of
+# learning from an imbalanced dataset, we will increase its ratio to 100:1
+
+import numpy as np
+import pandas as pd
```

Comment:
Since it will be a notebook style example, I feel it more friendly to have the import close to their first usage in the contrary to script style example.